packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -235,59 +263,11 @@ export class TeamsBotV2Impl {
     throw new Error("Invalid programming language");
   }
 
-  private resolveActRoles(ctx: Context, inputs: Inputs): PluginActRoles[] {
-    let actRoles: PluginActRoles[] = [];
-
-    const solutionSettings = ctx.projectSetting.solutionSettings as AzureSolutionSettings;
-    // Null check solutionSettings
-    const capabilities = solutionSettings.capabilities;
-    if (capabilities?.includes(PluginActRoles.Bot)) {
-      const scenarios = inputs?.[AzureSolutionQuestionNames.Scenarios];
-      if (isBotNotificationEnabled() && Array.isArray(scenarios)) {
-        const scenarioActRoles = scenarios
-          .map((item) => QuestionBotScenarioToPluginActRoles.get(item))
-          .filter((item): item is PluginActRoles => item !== undefined);
-        // dedup
-        actRoles = [...new Set([...actRoles, ...scenarioActRoles])];
-      } else {
-        // for legacy bot
-        actRoles.push(PluginActRoles.Bot);
-      }
-    }
-
-    if (capabilities?.includes(PluginActRoles.MessageExtension)) {
-      actRoles.push(PluginActRoles.MessageExtension);
-    }
-
-    return actRoles;
-  }
-
-  private resolveHostType(ctx: Context): HostType {
+  private resolveServiceType(ctx: Context): ServiceType {
     const rawHostType = ctx.projectSetting?.pluginSettings?.[PluginBot.PLUGIN_NAME]?.[
       PluginBot.HOST_TYPE
     ] as string;
-
-    switch (rawHostType) {
-      case "app-service":
-        return HostType.AppService;
-      case "azure-functions":
-        return HostType.Function;
-    }
-    throw new Error("Invalid host type");
-  }
-
-  private resolveTriggers(inputs: Inputs): BotTrigger[] {
-    const rawHostTypeTriggers = inputs?.[QuestionNames.BOT_HOST_TYPE_TRIGGER];
-    if (!Array.isArray(rawHostTypeTriggers)) {
-      return [];
-    }
-    // convert HostTypeTrigger question to trigger name
-    return rawHostTypeTriggers
-      .map((hostTypeTrigger) => {
-        const option = HostTypeTriggerOptions.find((option) => option.id === hostTypeTrigger);
-        return option?.trigger;
-      })
-      .filter((item): item is BotTrigger => item !== undefined);
+    return hostServiceTypeMapping[rawHostType];
```

Comment:
We need to add some error handling here, e.g. what if rawHostType is not defined in the mapping to make code robust.

Reply:
Added.